_schemaV2.graphql

```diff
@@ -12576,13 +12586,14 @@ type NotificationEdge {
 }
 
 type NotificationPreference {
-  channel: String!
+  channel: NotificationChannel!
```

Comment:
Did field get used by anything previously?

Reply:
`channel` was described when this was created but it wasn't being utilized until now

Reply:
on the gravity side of things it gets carried along with the rest of the data payload for the subscription group to be sent to braze: https://github.com/artsy/gravity/blob/main/spec/api/v1/notification_preferences_endpoint_spec.rb#L11 

but braze only handles email updates for the time being so it wasn't being utilized to switch between logic branches or anything.

Reply:
Yeah, we'd only be concerned with any frontend queries/mutations that might have used this.

Reply:
there are some areas outside of the preferences app that uses channel, but as along as` channel === "email` where its relevant it shouldn't break anything, but fingers crossed!

Reply:
Not exactly. This is a breaking schema change so the types will break.

Reply:
in the case of breaking, then it would have to be reverted back to a string instead of an enum??

Reply:
And for what it's worth: this is a change worth making — this should have never been `string` but we need to actually verify that nothing was using it previously, otherwise we'll have to deprecate and replace.

Reply:
the primary place where this would be a problem would be in SavedSearchAlerts, from what I see

Reply:
This is the other place that I see channel being utilized in both eigen & force: https://github.com/artsy/force/blob/main/src/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx#L94

Reply:
I'd check to see if this is used in Eigen before replacing it because then it gets much harder to deal with. But basically: we'd add a `deprecationReason` to this, add a new enum field, and replace/deploy.

Reply:
Sigh. So yeah. Shame but the path of least resistance here might be to just leave this alone as a string, document the supported types in the description and hope it doesn't happen again.

Reply:
its a copy/paste from force in eigen so yes: https://github.com/artsy/eigen/blob/main/src/app/Scenes/SavedSearchAlert/EditSavedSearchAlert.tsx#L44

Reply:
@jonallured would love your thoughts on this!

Reply:
Yeah, I think the best idea is to: run Metaphysics locally and sync your local schema with both Eigen and Force to determine if there are type errors.

Reply:
Will have to locally schema match MP to make test queries on eigen and force and that's tbd so this issue is currently unresolved.

Reply:
okay so after locally matching MP schema w/ force I did discover type errors to SavedSearchAlerts so the best course of action (and the easiest thing to do) is to revert channel back into a string. I added some descriptions for acceptable string values and included a comment about this in the code. I think we can call this conversation resolved btw.